Userland/Games/2048/Game.cpp

```diff
@@ -198,6 +219,72 @@ Game::MoveOutcome Game::attempt_move(Direction direction)
     return MoveOutcome::InvalidMove;
 }
 
+void Game::add_evil_tile()
+{
+    size_t worst_row = 0;
+    size_t worst_column = 0;
+    u32 worst_value = 2;
+
+    size_t most_free_cells = m_grid_size * m_grid_size; // a big value to min() against
+
+    // Largest theoretically possible score after a move.
+    // All the cells are filled with the same tile, so after a move every tile merges.
+    size_t worst_score = m_score + largest_tile() * m_grid_size * m_grid_size;
+
+    bool stalled = false;
+
+    for (size_t row = 0; row < m_grid_size && !stalled; row++) {
+        for (size_t column = 0; column < m_grid_size && !stalled; column++) {
+            if (m_board[row][column] != 0)
+                continue;
+
+            for (u32 value : Array { 2, 4 }) {
+                Game saved_state = *this;
+                saved_state.m_board[row][column] = value;
+
+                if (is_stalled(saved_state.m_board)) {
+                    stalled = true;
+
+                    worst_row = row;
+                    worst_column = column;
+                    worst_value = value;
+
+                    break;
+                }
+
+                size_t best_outcome = 0;
+                size_t best_score = 0;
```

Comment:
A comment explaining why we are suddenly looking for *best* outcome/score instead of the *worst* ones would help.

E.g.,

```cpp
// Look for the *best* score the player can achieve in one move stating from this state.
// We want this best score to be as low as possible.
```